### PR TITLE
feat: java sdk add api getTableSchema

### PR DESCRIPTION
**Pull request by lotabout**
Java SDK: add API for fetching table schema.

### CODE REVIEW

File: java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/SqlClusterExecutor.java [hunk 1]
```diff
@@ -220,6 +220,26 @@ public Schema getInputSchema(String dbName, String sql) throws SQLException {
...
 
+    @Override
+    public Schema getTableSchema(String dbName, String tableName) throws SQLException {
+        com._4paradigm.openmldb.Schema schema = sqlRouter.ShowTableSchema(dbName, tableName);
+        if (schema == null) {
+            throw new SQLException(String.format("table %s not found in db %s", tableName, dbName));
+        }
+
+        List<Column> columnList = new ArrayList<>();
+        for (int i = 0; i < schema.GetColumnCnt(); i++) {
```
Comment (dl239) [inline java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/SqlClusterExecutor.java]: use Common.convertSchema

File: src/sdk/sql_router.h [hunk 2]
```diff
@@ -131,6 +131,8 @@ class SQLRouter {
     virtual std::shared_ptr<openmldb::sdk::QueryFuture> CallSQLBatchRequestProcedure(
         const std::string& db, const std::string& sp_name, int64_t timeout_ms,
         std::shared_ptr<openmldb::sdk::SQLRequestRowBatch> row_batch, hybridse::sdk::Status* status) = 0;
+
+    virtual std::shared_ptr<hybridse::sdk::Schema> ShowTableSchema(const std::string& db, const std::string& table_name) = 0;
```
Comment (dl239) [inline src/sdk/sql_router.h]: rename as `GetTableSchema`

File: src/sdk/sql_cluster_router.h [hunk 3]
```diff
@@ -148,6 +148,8 @@ class SQLClusterRouter : public SQLRouter {
     std::shared_ptr<::openmldb::client::TabletClient> GetTabletClient(const std::string& db, const std::string& sql,
                                                                       const std::shared_ptr<SQLRequestRow>& row);
 
+    std::shared_ptr<hybridse::sdk::Schema> ShowTableSchema(const std::string& db, const std::string& table_name);
```
Comment (dl239) [inline src/sdk/sql_cluster_router.h]: add `override`

File: src/sdk/sql_cluster_router.cc [hunk 4]
```diff
@@ -1294,5 +1294,21 @@ std::shared_ptr<openmldb::sdk::QueryFuture> SQLClusterRouter::CallSQLBatchReques
...
+        return std::shared_ptr<hybridse::sdk::Schema>();
+    }
+
+    ::hybridse::vm::Schema output_schema;
+    if (::openmldb::catalog::SchemaAdapter::ConvertSchema(table_info->column_desc(), &output_schema)) {
+        return std::make_shared<::hybridse::sdk::SchemaImpl>(output_schema);
+    } else {
+        LOG(ERROR) << "Failed to convert schema for " + table_name + "in db " + db;
+    }
+    return std::shared_ptr<hybridse::sdk::Schema>();
```
Comment (dl239) [inline src/sdk/sql_cluster_router.cc]: add test. refer `src/sdk/sql_cluster_test.cc`

File: src/sdk/sql_cluster_router.cc [hunk 5]
```diff
@@ -1294,5 +1294,21 @@ std::shared_ptr<openmldb::sdk::QueryFuture> SQLClusterRouter::CallSQLBatchReques
     return future;
 }
 
+std::shared_ptr<hybridse::sdk::Schema> SQLClusterRouter::ShowTableSchema(const std::string& db, const std::string& table_name) {
+    auto table_info = cluster_sdk_->GetTableInfo(db, table_name);
+    if (!table_info) {
+        LOG(ERROR) << "table with name " + table_name + " in db " + db + " does not exist";
+        return std::shared_ptr<hybridse::sdk::Schema>();
```
Comment (vagetablechicken) [inline src/sdk/sql_cluster_router.cc]: You can use `return {};` to simplify it.

File: src/sdk/sql_cluster_router.cc [hunk 6]
```diff
@@ -1294,5 +1294,21 @@ std::shared_ptr<openmldb::sdk::QueryFuture> SQLClusterRouter::CallSQLBatchReques
...
+    }
+
+    ::hybridse::vm::Schema output_schema;
+    if (::openmldb::catalog::SchemaAdapter::ConvertSchema(table_info->column_desc(), &output_schema)) {
+        return std::make_shared<::hybridse::sdk::SchemaImpl>(output_schema);
```
Comment (vagetablechicken) [inline src/sdk/sql_cluster_router.cc:1306]: make a local var shared?

Comment (vagetablechicken) [reply src/sdk/sql_cluster_router.cc:1306]: > make a local var shared?

SchemaImpl ctor, got it.

**Review by dl239 — APPROVED** [review]

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**